packages/realm/bindgen/src/templates/typescript.ts

```diff
@@ -216,7 +205,7 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
 
   out("// Opaque types (including Key types)");
   for (const { jsName } of (spec.opaqueTypes as NamedType[]).concat(spec.keyTypes)) {
-    out.lines("/** Using an empty enum to express a nominal type */", `export declare enum ${jsName} {}`);
+    out.lines("/** Using an empty enum to express a nominal type */", `export enum ${jsName} {}`);
```

Comment:
```suggestion
    out.lines("/** Using an empty enum to express a nominal type */", `export const enum ${jsName} {}`);
```

We should probably consider some of the alternative mechanisms we discovered for opaque types at some point. But at the very least, we shouldn't imply that there is any actual value with that name.

Reply:
You mean like [we already do for classes](https://github.com/realm/realm-js/blob/63c2fcb315f95fa89ecf9daccf4755a9d973f9bc/packages/realm/bindgen/src/templates/typescript.ts#L256-L257)? Yes, possibly something like that. I wish there was a way to tell TS to model the behavior of `Object.create(null)` correctly, so that we could create a class that really had no known properties, but that doesn't seem possible right now.